src/ui/public/state_management/state_monitor.js

```diff
@@ -8,9 +8,14 @@ function stateMonitor(state, defaultState) {
   let destroyed = false;
   let ignoredProps = [];
   let changeHandlers = [];
+  let originalState;
 
-  // state.toJSON returns a reference, clone so we can mutate it safely
-  const originalState = cloneDeep(defaultState) || cloneDeep(state.toJSON());
+  setOriginalState(defaultState);
+
+  function setOriginalState(defaultState) {
```

Comment:
`defaultState` here is the initial value (optionally) passed in. Would `initialState` be clearer? I'm not sure how else to clean this up.